src/pages/workspaces/workspace/workflows/WorkflowView.js

```diff
@@ -395,11 +395,17 @@ const WorkflowView = _.flow(
     // savedConfig: unmodified copy of config for checking for unsaved edits
     // modifiedConfig: active data, potentially unsaved
     const {
-      isFreshData, savedConfig, entityMetadata, launching, activeTab, useCallCache, deleteIntermediateOutputFiles,
+      currentSnapRedacted, isFreshData, savedConfig, entityMetadata, launching, activeTab, useCallCache, deleteIntermediateOutputFiles,
       entitySelectionModel, variableSelected, modifiedConfig, updatingConfig
     } = this.state
+    const { methodVersion, methodNamespace, methodName, methodPath, sourceRepo } = modifiedConfig.methodRepoMethod
     const { namespace, name, workspace } = this.props
     const workspaceId = { namespace, name }
+    const workflowMetadata = {
+      // Source follows the logic of the rendering on the page. It is not identical because the rendering requires elements to be included.
+      snapshotId: modifiedConfig.methodRepoMethod.methodVersion, source: sourceRepo === 'agora' || currentSnapRedacted ?
```

Comment:
I also noticed I had to build the source by hand. It wasn't quite reusable/duplicated code, so I didn't duplicate it, but that may be my inexperience with hyperscript. I'd love some eyes on that too.